src/coreclr/nativeaot/Runtime/thread.cpp

```diff
@@ -1346,6 +1361,16 @@ COOP_PINVOKE_HELPER(uint64_t, RhCurrentOSThreadId, ())
     return PalGetCurrentThreadIdForLogging();
 }
 
+COOP_PINVOKE_HELPER(int32_t, RhGetCurrentManagedThreadId, ())
```

Comment:
Once/if we do the codegen work to make thread statics fast, this change will be a de-optimization and we will go back to managed thread static.

Reply:
Yes, once threadstatics are faster, we can just use a threadstatic.

Reply:
I am thinking that we should keep using regular threadstatics for this even if it means that we are leaving some perf on the table for now.

Reply:
Using threadstatics makes the lock noticeable slower. 

Straightforward way with `IdNone == 0` and without `CurrentManagedThreadIdUnchecked`  is  36% slower.
-1 adjusted way with `IdNone == -1` and `CurrentManagedThreadIdUnchecked` is 33% slower

```
=== current:
Fat:
8031
Thin:
5684
Fat:
8034
Thin:
5680
Fat:
8000
Thin:
5810

=== use threadstatic without CurrentManagedThreadIdUnchecked

thin lock gets ~36% slower

Fat:
9122
Thin:
7796
Fat:
9135
Thin:
7766
Fat:
9119
Thin:
7813

=== use threadstatic with CurrentManagedThreadIdUnchecked and -1 
adjustment so that uninitialized looks like -1 

thin lock gets 33% slower

Fat:
8759
Thin:
7567
Fat:
8747
Thin:
7583
Fat:
8713
Thin:
7574
```

Reply:
I am going to switch to the `threadstatic with CurrentManagedThreadIdUnchecked and -1 ` variant then. 

It is still better than before and a possible gain of at least 33% should motivate us to improve threadstatics.